src/internal/obj/factory.go

```diff
@@ -418,36 +417,45 @@ type ObjectStoreURL struct {
 	Bucket string
 	// The object itself.
 	Object string
+	// The query parameters if defined.
+	Params string
 }
 
 func (s ObjectStoreURL) String() string {
 	return fmt.Sprintf("%s://%s/%s", s.Scheme, s.Bucket, s.Object)
 }
 
+func (s ObjectStoreURL) BucketString() string {
+	bucket := fmt.Sprintf("%s://%s", s.Scheme, s.Bucket)
+	if s.Params != "" {
+		bucket += "?" + s.Params
+	}
+	return bucket
+}
+
 // ParseURL parses an URL into ObjectStoreURL.
 func ParseURL(urlStr string) (*ObjectStoreURL, error) {
 	u, err := url.Parse(urlStr)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error parsing url %v", urlStr)
 	}
 	switch u.Scheme {
-	case "s3", "gcs", "gs", "local":
+	case "azblob", "gcs", "gs", "local", "s3":
```

Comment:
Is `gs` valid?

Reply:
It looks like `gs` is the standard scheme and I cannot find a reference to `gcs`.

Reply:
Basically, I don't think we should be supporting schemes that are not actually supported by the backend. If we could do that replacement in one place and only operate on valid schemes afterwards, that would be ideal.